source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h

```diff
@@ -21,15 +22,26 @@ class LocalRateLimiterImpl {
 
   bool requestAllowed() const;
 
+  void fillDescriptor(const std::vector<Envoy::RateLimit::Descriptor>& descriptors);
+
+  bool requestAllowed(const Envoy::RateLimit::Descriptor& descriptor) const;
+
 private:
   void onFillTimer();
+  void onFillTimerDescriptorHelper();
 
   const std::chrono::milliseconds fill_interval_;
   const uint32_t max_tokens_;
   const uint32_t tokens_per_fill_;
   const Event::TimerPtr fill_timer_;
   mutable std::atomic<uint32_t> tokens_;
   mutable Thread::ThreadSynchronizer synchronizer_; // Used for testing only.
+  struct DescriptorToken {
+    mutable std::atomic<uint32_t> tokens;
+  };
+  absl::flat_hash_map<Envoy::RateLimit::Descriptor, std::unique_ptr<DescriptorToken>>
+      tokens_per_descriptor_;
+  absl::flat_hash_map<Envoy::RateLimit::Descriptor, long int> last_refill_per_descriptor_;
```

Comment:
```suggestion
  absl::flat_hash_map<Envoy::RateLimit::Descriptor, uint32_t> last_refill_per_descriptor_;
```

Reply:
This will definitely change now as we would be using tokens only...

Reply:
Thanks..fixed

Reply:
Left as `long int` because otherwise we loose precision of `count` of std::chrono::milliseconds